ts/Dashboards/Plugins/NavigatorComponent.ts

```diff
@@ -665,20 +670,77 @@ class NavigatorComponent extends Component {
             const table = this.connector.table,
                 options = this.options,
                 column = this.getColumnAssignment(),
-                values = (table.getColumn(column[0], true) || []);
+                columnValues = table.getColumn(column[0], true) || [],
+                crossfilterOptions = options.sync.crossfilter;
 
-            let data: (
-                Array<(number|string|null)>|
-                Array<[(number|string), number]>
-            );
+            let values: DataTable.Column = [],
+                data: (
+                    Array<(number|string|null)>|
+                    Array<[number|string, number|null]>
+                );
 
-            if (options.sync.crossfilter) {
-                const seriesData: Array<[(number|string), number]> = [],
-                    xData: Array<(number|string)> = [];
+            if (crossfilterOptions === true || (
+                typeof crossfilterOptions === 'object' &&
+                crossfilterOptions.enabled
+            )) {
 
-                let index: number;
+                const seriesData: Array<[(number|string), number]> = [],
+                    xData: Array<(number|string)> = [],
+                    modifierOptions = table.getModifier()?.options;
+
+                let index: number,
+                    max: number|undefined = void 0,
+                    min: number|undefined = void 0;
+
+                if (
+                    crossfilterOptions !== true &&
+                    crossfilterOptions.affectNavigator &&
+                    modifierOptions?.type === 'Range'
+                ) {
+                    const appliedRanges =
+                        (modifierOptions as RangeModifierOptions).ranges.filter(
+                            (range): boolean => range.column !== column[0]
+                        ),
+                        rangedColumns = appliedRanges.map(
+                            (range): DataTable.Column => (
+                                table.getColumn(range.column, true) || []
+                            )
+                        );
+
+                    for (let i = 0; i < columnValues.length; i++) {
```

Comment:
Replace with more efficent `for(let i = 0, iEnd = columnValues.length; i < iEnd; ++i)`